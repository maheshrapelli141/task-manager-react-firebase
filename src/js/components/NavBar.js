import React from 'react';
import { connect } from 'react-redux';

import NavItem from './NavItem';
import { auth, provider } from '../util/firebase';

const login = () => { auth.signInWithPopup(provider) }
const logout = () => { auth.signOut() }

const NavBar = props => (
  <nav className="navbar navbar-expand-lg navbar-dark bg-primary">
    <a className="navbar-brand" href="#">
      Navbar
    </a>
    <button
      className="navbar-toggler"
      type="button"
      data-toggle="collapse"
      data-target="#navbarContent"
    >
      <span className="navbar-toggler-icon" />
    </button>

    <div className="collapse navbar-collapse" id="navbarContent">
      {props.user
        ?
          <ul className="navbar-nav mr-auto">
            <NavItem path="/task" label="Task" />
            <NavItem path="/report" label="Report" />
            <button type="button" className="btn btn-primary btn-right" onClick={logout}  >Logout</button>
          </ul>
        :
          <ul className="navbar-nav mr-auto">
            <button type="button" className="btn btn-primary btn-right" onClick={login}  >Login</button>
          </ul>
      }

    </div>
  </nav>
);

const mapStateToProps = state => {
  const { user } = state;
  return {
    user
  };
}

export default connect(mapStateToProps)(NavBar);
