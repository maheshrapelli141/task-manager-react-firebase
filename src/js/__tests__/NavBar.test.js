import React from 'react';
import { shallow } from 'enzyme';

import NavBar from '../components/NavBar';

const wrapper = shallow(<NavBar />);

describe('NavBar component', () => {
  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
