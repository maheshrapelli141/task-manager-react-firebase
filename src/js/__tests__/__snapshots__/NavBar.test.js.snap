// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component should render correctly 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-dark bg-primary"
>
  <a
    className="navbar-brand"
    href="#"
  >
    Navbar
  </a>
  <button
    className="navbar-toggler"
    data-target="#navbarContent"
    data-toggle="collapse"
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    className="collapse navbar-collapse"
    id="navbarContent"
  >
    <ul
      className="navbar-nav mr-auto"
    >
      <NavItem
        label="Task"
        path="/task"
      />
      <NavItem
        label="Report"
        path="/report"
      />
    </ul>
  </div>
</nav>
`;
